osu.Game/Overlays/BeatmapSet/BeatmapBadge.cs

```diff
@@ -14,33 +14,33 @@
 
 namespace osu.Game.Overlays.BeatmapSet
 {
-    public abstract class BeatmapBadgePill : CompositeDrawable
+    public abstract class BeatmapBadge : CompositeDrawable
     {
-        [Resolved]
-        protected OsuColour Colours { get; private set; } = null!;
-
-        [Resolved(canBeNull: true)]
-        protected OverlayColourProvider? ColourProvider { get; private set; }
-
         /// <summary>
         /// The text displayed on the badge's label.
         /// </summary>
-        public abstract LocalisableString BadgeText { get; }
+        public LocalisableString BadgeText
+        {
+            set => badgeLabel.Text = value.ToUpper();
```

Comment:
It should be safe to call this setter at any time, and as written it isn't, because if it is attempted to set this prior to BDL, it will fail with a nullref. Same goes for `BadgeColour`.

Reply:
If that's a concern, think it's better moving declaration to constructor instead.

Reply:
Or just do the usual `IsLoaded` song and dance.. Either-or is ok with me.